mlflow/data/__init__.py

```diff
@@ -12,6 +13,10 @@
 from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
 from mlflow.utils.annotations import experimental
 
+with suppress(ImportError):
+    # Suppressing ImportError to pass mlflow-skinny testing.
+    from mlflow.data import meta_dataset  # noqa: F401
```

Comment:
what part of `meta_dataset` is breaking the test?

Reply:
i see. it looks like we handle these in the dataset registry `mlflow/data/dataset_registry.py` rather than in the module `__init__.py`. can we use that approach here for consistency?

Reply:
Actually those imports inside `mlflow/data/dataset_registry.py` should also be written in `__init__.py`, otherwise it's quite unclear why `mlflow.data.numpy_dataset` is a valid module without having `from mlflow.data import numpy_dataset` in the `__init__.py`. I will open a followup PR to clean them up.